vcf-dumper/vcf-dumper-ws/src/main/java/uk/ac/ebi/eva/vcfdump/server/rest/HtsgetVcfController.java

```diff
@@ -53,145 +56,182 @@
-        return ResponseEntity.status(HttpStatus.OK).body(Collections.singletonMap("htsget",  htsGetResponse));
+        return ResponseEntity.status(HttpStatus.OK).body(Collections.singletonMap("htsget", htsGetResponse));
     }
 
-    private ResponseEntity validateRequest(String referenceName, Long start, VariantExporterController controller) {
-        if (!controller.validateSpecies()) {
-            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(
-                    Collections.singletonMap("htsget", new HtsGetError("InvalidInput", "The requested species is not available")));
+    private Optional<ResponseEntity> validateParameters(String format, String referenceName, Long start, Long end) {
+        if (!VCF.equals(format)) {
+            return Optional.of(getResponseEntity("UnsupportedFormat",
+                                                 "The requested file format is not supported by the server",
+                                                 HttpStatus.BAD_REQUEST));
         }
-        if (!controller.validateStudies()) {
-            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(
-                    Collections.singletonMap("htsget", new HtsGetError("InvalidInput", "The requested study(ies) is not available")));
+        if (start != null && end != null && end < start) {
+            return Optional.of(getResponseEntity("InvalidRange", "The requested range cannot be satisfied",
+                                                 HttpStatus.BAD_REQUEST));
         }
         if (start != null && referenceName == null) {
-            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(
-                    Collections.singletonMap("htsget", new HtsGetError("InvalidInput", "Reference name is not specified when start is specified")));
+            return Optional.of(getResponseEntity("InvalidInput", "Reference name is not specified when start is " +
+                    "specified", HttpStatus.BAD_REQUEST));
         }
-        return null;
+        if (referenceName == null) {
+            return Optional.of(getResponseEntity("Unsupported", "'referenceName' is required", HttpStatus.BAD_REQUEST));
+        }
+        return Optional.empty();
+    }
+
+    private ResponseEntity getResponseEntity(String error, String message, HttpStatus httpStatus) {
+        HtsGetError htsGetError = new HtsGetError(error, message);
+        return ResponseEntity.status(httpStatus).body(Collections.singletonMap("htsget", htsGetError));
+    }
+
+    private Optional<ResponseEntity> validateRequest(String referenceName, Long start, Long end,
+                                                     VariantExporterController controller) {
+        if (start == null) {
+            start = controller.getCoordinateOfFirstVariant(referenceName);
+        }
+        if (end == null) {
+            end = controller.getCoordinateOfLastVariant(referenceName);
+        }
```

Comment:
start/end null checks are already being done before the method invocation, so not needed here...